clients/privacy-center/app/server-environment.ts

```diff
@@ -0,0 +1,210 @@
+ */
+export interface PrivacyCenterEnvironment {
+  fidesApiUrl: string;
+  config?: Config;
+  styles?: string;
+}
+
+/**
+ * Load a config file from the given list of URLs, trying them in order until one is successfully read.
+ */ 
+const loadConfigFile = async (urls: (string | undefined)[]): Promise<string | undefined> => {
+  // Dynamically import the "fs" module to read from the filesystem. This module
+  // doesn't exist in the browser context, so to allow the bundler to function
+  // we provide a (non-functional) fallback for "fs" in the webpack config (see
+  // next.config.js)
+  const fsPromises = (await import("fs")).promises;
+  if (!fsPromises) {
+    throw new Error("Unable to load 'fs' module!");
+  }
+
+  // Loop through the provided URLs, testing each one in order, and return the
+  // first file that loads.
+  /* eslint-disable no-restricted-syntax,no-continue,no-await-in-loop */
+  for (const urlString of urls) {
+    try {
+      if (!urlString) {
+        continue;
+      }
+      const url: URL = new URL(urlString);
+      // DEFER: add support for https:// to fetch remote config files!
+      if (url.protocol !== "file:") {
+        throw new Error(`Config file URLs currently must use the 'file:' protocol: ${urlString}`);
+      }
+      // Relative paths (e.g. "file:./") aren't supported by node's URL class.
+      // So to support this, we just use a path string instead!
+      let path;
+      if (urlString.startsWith("file:.")) {
+         path = urlString.replace("file:", "");
+      }
+      const file = await fsPromises.readFile(path || url, "utf-8");
+      console.log(`Loaded configuration file: ${urlString}`);
+      return file;
+    } catch (err: any) {
+      // Catch "file not found" errors (ENOENT)
+      if (err.code === "ENOENT") {
+        continue;
+      }
+      // Log everything else and continue
+      console.log(`Failed to load configuration file from ${urlString}. Error: `, err);
+    }
+  }
+  /* eslint-enable no-restricted-syntax,no-continue,no-await-in-loop */
+}
+
+/**
+ * Load the config.json file from the given URL, or fallback to default filesystem paths.
+ * 
+ * Loading precedence is:
+ * 1) Load from `configJsonUrl` argument
+ * 2) Load from file:///app/config/config.json on filesystem (absolute path)
+ * 3) Load from file:./config/config.json on filesystem (relative path)
+ * 
+ * NOTE: The "/app/config" path is for backwards-compatibility with the initial
+ * version of ethyca/fides-privacy-center, which expected to always load the
+ * configuration file from this well-known path.
+ */
+export const loadConfigFromFile = async (configJsonUrl?: string): Promise<Config | undefined> => {
+  const urls = [
+    configJsonUrl,
+    "file:///app/config/config.json",
+    "file:./config/config.json",
+  ];
+  const file = await loadConfigFile(urls);
+  if (file) {
+    const config = JSON.parse(file) as Config;
+    // TODO: validate config?
```

Comment:
[Zod](https://github.com/colinhacks/zod) would be helpful to validate this. It's essentially pydantic but for typescript. It's super powerful! 


This might helpful too. There is a package called [t3-env](https://github.com/t3-oss/t3-env) that will uses a zod schema to validate env vars on startup. If any are missing or invalid it will fail startup and let the user know what's missing.